stage1/init/common/path.go

```diff
@@ -65,7 +65,7 @@ func ServiceWantPath(root string, appName types.ACName) string {
 func InstantiatedPrepareAppUnitName(appName types.ACName) string {
 	// Naming respecting escaping rules, see systemd.unit(5) and systemd-escape(1)
 	escapedRoot := unit.UnitNamePathEscape(common.RelAppRootfsPath(appName))
-	return "prepare-app@" + escapedRoot + ".service"
+	return "prepare-app@-" + escapedRoot + ".service"
```

Comment:
Yes: the string after the `@` is not part of the filename written on disk but the parameter to the parametrized systemd unit. In this case, it is a directory, for example `/opt/stage2/busybox/rootfs`. But the string is escaped: `/` are converted to `-`, following systemd escaping rules. And the variable `escapedRoot` is not an absolute path: it does not have the leading `/`. It was not a problem before because `prepare-app` was accepting a relative path as parameter. But I made changes below that requires an absolute path. Hence this fix.